### PR TITLE
Enable using filters, fields, exclude_fields for a QueryFile object.

### PR DESCRIPTION
**Pull request by nikolamaric**

### CODE REVIEW

**Comment by nikolamaric** [comment]
closes #383

File: solvebio/query.py
Comment (davecap) [inline solvebio/query.py:1192]: Agree, we can remove from here

**Review by davecap — APPROVED** [review]
Lets wait to get the tests passing before merging this

File: solvebio/query.py
Comment (davecap) [inline solvebio/query.py:1150]: It is probably ok to just print the contents of [total] as-is, since it is a debug log.

Comment (nikolamaric) [reply solvebio/query.py]: Done.

File: solvebio/query.py
Comment (davecap) [inline solvebio/query.py:1024]: set limit to inf

File: solvebio/query.py
Comment (davecap) [inline solvebio/query.py:267]: I guess the issue here is that because we don't have `total`, it tries to send `inf` to the API and fails somewhere? So we need to set a large value for limit here. Are there any other options instead of setting to 10**15?

Comment (davecap) [reply solvebio/query.py:267]: What happens if we use  `None`?

Comment (nikolamaric) [reply solvebio/query.py:267]: for example `__len__` is called here: https://github.com/solvebio/solvebio-python/blob/42e8aab4334c9bc71a88744b7f67db8bc3ea198e/solvebio/query.py#L413

and `None` raises an error `TypeError: 'NoneType' object cannot be interpreted as an integer`.

Honestly, I do not have an idea of what we can add there instead of some large value?

Comment (davecap) [reply solvebio/query.py:267]: What if we handle both cases where total and _limit are both None in the iterator, instead of wrapping it up in `len(self)`?

Comment (nikolamaric) [reply solvebio/query.py:267]: It means that we have to do the same wherever `len(self)` is currently called in code, right?

Comment (davecap) [reply solvebio/query.py:267]: What if in __len__ we `return None if _limit is None and self.count() is None`

Comment (nikolamaric) [reply solvebio/query.py:267]: it is possible, but as far as I know instead of calling `len(self)` we have to call it like `self.__len__()` otherwise it will raise `TypeError: 'NoneType' object cannot be interpreted as an integer`

Comment (davecap) [reply solvebio/query.py:267]: Ok can you set this magic value as a constant in the function `INF = 10**15` and add a comment above it to explain

Comment (davecap) [reply solvebio/query.py]: Actually, last thing. Can we just set the initial value of limit to INF instead of float(inf)?

**Comment by davecap** [comment]
Great, will merge when tests pass!

**Comment by jsh2134** [comment]
😻

**Pull request merged**
